src/main/java/eu/europeana/metis/sandbox/controller/DatasetController.java

```diff
@@ -421,8 +421,10 @@ public List<LanguageView> getAllLanguages() {
     @ApiResponse(responseCode = "400", description = MESSAGE_FOR_400_CODE)
     @PostMapping(value = "{id}/debias", produces = APPLICATION_JSON_VALUE)
     @ResponseStatus(HttpStatus.OK)
-    public boolean processDebias(@PathVariable("id") Integer datasetId) {
-        return debiasDetectService.process(datasetId);
+    public boolean processDeBias(@PathVariable("id") Integer datasetId) {
+        debiasStateService.cleanDeBiasReport(datasetId);
+        debiasStateService.setState(debiasStateService.getReady());
+        return debiasStateService.process(datasetId);
```

Comment:
Should we not check whether the dataset is already processed? Or currently processing? In either case, it's probably not a good idea to start again.

Reply:
good catch! I'll add the check.